rdmd.d

```diff
@@ -466,24 +457,18 @@ private int rebuild(string root, string fullExe,
         fullExe = fullExe.defaultExtension(".exe");
 
     // Delete the old executable before we start building.
-    yap("stat ", fullExe);
-    if (exists(fullExe))
+    if (Filesystem.exists(fullExe))
     {
-        enforce(!isDir(fullExe), fullExe ~ " is a directory");
-        yap("rm ", fullExe);
-        if (!dryRun)
+        enforce(Filesystem.isFile(fullExe), fullExe ~ " is not a regular file");
```

Comment:
This changes app logic: is `enforce(isFile)` definitely the same as `enforce(!isDir)` ... ?

Reply:
Not the same, this is a one-line bug fix.  Thought it would be better to include in this review rather than creating a new PR for it, since it's "somewhat" related in that it has to change because of this PR anyway.

Reply:
> Not the same, this is a one-line bug fix.

I would suggest making that fix in its own separate PR, in line with the contribution guidelines:
https://github.com/dlang/tools/pull/299#issuecomment-360219131

I'd also suggest that that fix be checked _carefully_ to make sure that it really does reflect the correct/wanted behaviour.